apps/api/src/fcm/fcm.service.ts

```
@@ -69,16 +69,43 @@ export class FcmService {
         ({ fcmTokenId }) => fcmTokenId,
       );
     }
+    const batches = tokens.reduce((acc, token, index) => {
+      if (index % 500 === 0) return [[token], ...acc];
+      const [first, ...array] = acc;
+      return [[...first, token], ...array];
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)